File: virtual time capsule/frontend/virtual-time-capsule-ui/src/app/dtos/friendship.dto.ts
import { FriendshipStatus } from '../enums/friendship-status.enum'; // Ensure this path is correct

export interface FriendshipDto {
  id: number;
  requesterId: number;
  requesterUsername: string;
  responderId: number;
  responderUsername: string;
  status: FriendshipStatus;
  lastUpdate: string; // ISO string
}

File: virtual time capsule/backend/src/main/java/com/java/web/virtual/time/capsule/dto/FriendshipDto.java
package com.java.web.virtual.time.capsule.dto;

import com.java.web.virtual.time.capsule.enums.FriendshipStatus; 
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class FriendshipDto {
    private Long id;
    private Long requesterId; 
    private String requesterUsername; 
    private Long responderId; 
    private String responderUsername; 
    private FriendshipStatus status;
    private String lastUpdate; 
}
